core/vulkan/vk_virtual_swapchain/cc/virtual_swapchain.cpp

```diff
@@ -294,6 +305,64 @@ void VirtualSwapchain::Destroy(const VkAllocationCallbacks* pAllocator) {
   functions_->vkDestroyCommandPool(device_, command_pool_, pAllocator);
 }
 
+void VirtualSwapchain::DumpImageToFile(uint8_t* image_data, size_t size) {
+  std::unique_ptr<char[]> image_data_owned(new char[size]());
+  memcpy(image_data_owned.get(), image_data, size);
+
+  auto now = std::chrono::system_clock::now().time_since_epoch().count();
+  auto image_path = image_dump_dir_ + "/image_" +
+                    std::to_string(dumped_frame_count_++) + "_ts_" +
+                    std::to_string(now) + ".ppm";
+
+  auto ppm_writer = [](std::unique_ptr<char[]> image_data, size_t size,
```

Comment:
Nit: It would be nice to see more empty lines breaking this up for readability,

Reply:
I'm not a fan of a big lambda to encapsulate the whole PPM encoder, could it be possible to have a separate function for it?

Or just use a wrapper around lodepng, if we go that way.